### ISSUE TITLE
Update search function to use Google cloud search API

### ISSUE THREAD

**Issue by stevenbird** [aikuma/aikuma#462]
@hleeldc please give details

**Comment by hleeldc** [comment]
There is a new class implementing the org.lp20.aikuma.storage.Index interface. The constructor looks like this.

```java
package org.lp20.aikuma.storage.google;

class GoogleDriveIndex2 implements Index (
    String root, /* name of the aikuma root directory */
    org.lp20.aikuma.storage.google.TokenManager tm,  /* provides accessToken(): String */
    String baseUrl,  /* server url */
    String idToken  /* identification token */
) {...}
```

The `root` parameter is the name of the central google drive that is shared with users. The app will have to request sharing of the central storage from the server. This sharing function will be almost the same as the existing file sharing, but it has yet to be implemented.

The `tm` parameter is a TokenManager object. TokenManager is an interface and instantiation via anonymous class is recommended.

```java
new TokenManager() {
    @Override
    public String accessToken() {
        return mCurrentAccessToken;
    }
}
```

For `baseUrl` and `idToken`, use the same values that were used with FusionIndex2.

---

Try to compile the app with cloud-storage 0.10.0, which currently lives on [my personal branch](/hleeldc/aikuma/tree/cloud-storage-0.10.0). There was a rearrangement of the package. For example, the GoogleDriveStorage and GoogleAuth moved into the google sub-directory (from org.lp20.aikuma.storage to org.lp20.aikuma.storage.google).

---

Before things function correctly, there are a few things to be done.

- The server is still using FusionIndex. The server code has been updated to use GoogleDriveIndex, but it still has to be tested and deployed.
- Also as mentioned earlier, a new web API function for sharing the central google drive.
- I also have to update the GoogleDriveStorage to undo the directory structure.

**Comment by stevenbird** [comment]
Resolved by #542
